### ISSUE TITLE
why AllenNlp character based Model does not give good prediction ?

### ISSUE THREAD

**Issue by elkhadirzyad** [allenai/allennlp#4765]
hello, In AllenNlp model, i tried to modify the forward method in order to predict the next caracter given a sentence, example for sentence = “distribution” 
i got "distribution of the regression model with a continuous variables@@UNKNOWN@@redect statistical test in r@@UNKNOWN@@repect the same da"

here is the forward() method 

`def forward(self, input_tokens, output_tokens):
        '''
        This is the main process of the Model where the actual computation happens. 
        Each Instance is fed to the forward method. 
        It takes dicts of tensors as input, with same keys as the fields in your Instance (input_tokens, output_tokens)
        It outputs the results of predicted tokens and the evaluation metrics as a dictionary. 
        '''

        mask = get_text_field_mask(input_tokens)
        embeddings = self.embedder(input_tokens)
        rnn_hidden = self.rnn(embeddings, mask)
        out_logits = self.hidden2out(rnn_hidden)
        
        loss = sequence_cross_entropy_with_logits(out_logits, output_tokens['tokens'], mask)
        
        #tensor = out_logits[:,out_logits.size()-1]
        #state = (torch.zeros(1, 1, self.hidden_size), torch.zeros(1, 1, self.hidden_size))
        words=[]
       
        output_t=out_logits[:,out_logits.size()[1]-1]
        
        test=torch.argmax(output_t).item()
        #if torch.argmax(output_t).item().size()[1]:
        try:
          token = Token(text=self.vocab.get_token_from_index(test, 'tokens'))
          words.append(token)
        except:
          print("An exception occurred could be ignored") 
       
        return {'loss': loss, 'nextCaracter': words}
`
 function predict() looks like 
`def predict(text: str, model: Model) -> float:
    tokenizer = CharacterTokenizer()
    tokens = tokenizer.tokenize(text)
    
    token_indexers = {'tokens': SingleIdTokenIndexer()}
    instance = tokens_to_instance(tokens, token_indexers)
    output = model.forward_on_instance(instance)
    print(output)`

**Comment by AkshitaB** [comment]
Hi @elkhadirzyad Can you specify the model's definition (init function)?

**Comment by elkhadirzyad** [comment]
@AkshitaB here is the init()
` def __init__(self,
                 embedder: TextFieldEmbedder,
                 hidden_size: int,
                 max_len: int,
                 vocab: Vocabulary) -> None:
        super().__init__(vocab)

        self.embedder = embedder

        # initialize a Seq2Seq encoder, LSTM
        self.rnn = PytorchSeq2SeqWrapper(
            torch.nn.LSTM(EMBEDDING_SIZE, HIDDEN_SIZE, batch_first=True))

        self.hidden2out = torch.nn.Linear(in_features=self.rnn.get_output_dim(), out_features=vocab.get_vocab_size('tokens'))
        self.hidden_size = hidden_size
        self.max_len = max_len`

**Comment by AkshitaB** [comment]
@elkhadirzyad Nothing pops out as an obvious issue with the code. Models do not always give perfect predictions! It may be helpful to train further, and judge the models based on metrics for downstream tasks.
